api/metrics/metrics.go

```diff
@@ -0,0 +1,48 @@
+package metrics
+
+import (
+	"github.com/ethereum-optimism/optimism/op-service/metrics"
+	opmetrics "github.com/ethereum-optimism/optimism/op-service/metrics"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type BlockIdType string
+
+var (
+	MetricsNamespace = "blob_api"
+
+	BlockIdTypeHash    BlockIdType = "hash"
+	BlockIdTypeBeacon  BlockIdType = "beacon"
+	BlockIdTypeInvalid BlockIdType = "invalid"
+)
+
+type Metricer interface {
+	Registry() *prometheus.Registry
+	RecordBlockIdType(t BlockIdType)
+}
+
+type metricsRecorder struct {
+	inputType *prometheus.CounterVec
```

Comment:
I'd disambiguate this a bit more, maybe name it "blockInputType" and add a comment on its possible values?